tests/e2e/e2e_ica_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// TestICARegister must run before any other
+func (s *IntegrationTestSuite) TestICA_1_Register() {
+	s.Run("register_ICA", func() {
+		connectionID := "connection-0"
+		var owner string
+		ownerAddr, err := s.chainA.genesisAccounts[1].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		owner = ownerAddr.String()
+		s.registerICA(owner, connectionID)
+
+		time.Sleep(2 * time.Minute)
+
+		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+		s.Require().Eventually(
+			func() bool {
+				icaAddr, err := queryICAaddr(chainAAPIEndpoint, owner, connectionID)
+				s.T().Logf("%s's interchain account on chain %s: %s", owner, s.chainB.id, icaAddr)
+				s.Require().NoError(err)
+				return owner != "" && icaAddr != ""
+			},
+			2*time.Minute,
+			10*time.Second,
+		)
+	})
+}
+
+func (s *IntegrationTestSuite) TestICA_2_BankSend() {
+	s.Run("test ica transactions", func() {
+		chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+		chainBAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainB.id][0].GetHostPort("1317/tcp"))
+		connectionID := "connection-0"
+		// step 1: get ica addr
+		icaOwnerAddr, err := s.chainA.genesisAccounts[1].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		icaOnwer := icaOwnerAddr.String()
+
+		var ica string
+		s.Require().Eventually(
+			func() bool {
+				ica, err = queryICAaddr(chainAAPIEndpoint, icaOnwer, connectionID)
+				s.Require().NoError(err)
+
+				return err == nil && ica != ""
+			},
+			time.Minute,
+			5*time.Second,
+		)
+
+		// step 2: fund ica, send tokens from chain b val to ica on chain b
+		senderAddr, err := s.chainB.validators[0].keyInfo.GetAddress()
+		s.Require().NoError(err)
+		sender := senderAddr.String()
+
+		s.sendMsgSend(s.chainB, 0, sender, ica, tokenAmount.String(), fees.String(), false)
+
+		s.Require().Eventually(
+			func() bool {
+				afterSenderICAbalance, err := getSpecificBalance(chainBAPIEndpoint, ica, uatomDenom)
+				s.Require().NoError(err)
+				return afterSenderICAbalance.IsEqual(tokenAmount)
+			},
+			time.Minute,
+			5*time.Second,
+		)
+
+		receiver := sender
+		var beforeICAsendReceiverbalance sdk.Coin
+		s.Require().Eventually(
+			func() bool {
+				beforeICAsendReceiverbalance, err = getSpecificBalance(chainBAPIEndpoint, receiver, uatomDenom)
+				s.Require().NoError(err)
+
+				return !beforeICAsendReceiverbalance.IsNil()
+			},
+			time.Minute,
+			5*time.Second,
+		)
+
+		// step 3: prepare ica bank send json
+		sendamt := sdk.NewCoin(uatomDenom, math.NewInt(1000000))
```

Comment:
should be some amount that is sure to be less than or equal to the `tokenAmount` that the account received already. We don't want this test breaking if the `tokenAmount` variable is changed to less than `1000000` one day.

maybe 1/10 of `tokenAmount` to make sure it's not too much but also not too little?

Reply:
sounds good !